### PR TITLE
Fix assertion in SparseMatrix

### PR DESCRIPTION
**Pull request by volkm**
Assertion was violated for empty matrix.

### CODE REVIEW

**Comment by sjunges** [comment]
Is it on purpose that I can now iterate over row 20 if there are no rows, but i cannot iterate over row 20 if there are 5 rows?

**Comment by volkm** [comment]
Good point. I revised the assertion.

File: src/storm/storage/SparseMatrix.cpp [hunk 1]
```diff
@@ -2382,22 +2382,25 @@ typename SparseMatrix<ValueType>::rows SparseMatrix<ValueType>::getRowGroup(inde
...
 
 template<typename ValueType>
 typename SparseMatrix<ValueType>::iterator SparseMatrix<ValueType>::begin(index_type row) {
+    STORM_LOG_ASSERT(row < this->getRowCount() || row == 0, "Row " << row << " exceeds row count " << this->getRowCount() << ".");
     return this->columnsAndValues.begin() + this->rowIndications[row];
 }
 
 template<typename ValueType>
 typename SparseMatrix<ValueType>::const_iterator SparseMatrix<ValueType>::end(index_type row) const {
+    STORM_LOG_ASSERT(row < this->getRowCount(), "Row " << row << " exceeds row count " << this->getRowCount() << ".");
```
Comment (sjunges) [inline src/storm/storage/SparseMatrix.cpp:2397]: for empty matrices, you cannot call end now?

**Comment by sjunges** [comment]
What is the use case to be able to iterate over a non-existing row?

**Comment by volkm** [comment]
In my view, it should be possible to create an empty matrix. But then the new test in this PR fails, because in the constructor we call `updateNonzeroEntryCount` which tries to iterate over all entries.
My goal is to support the iterators for empty matrices as well, such that for example `begin() != end()`  always works.

**Comment by sjunges** [comment]
I agree that begin() should work, but I dont know whether begin(0) needs to work?

**Comment by volkm** [comment]
I think we need to support 0 because it is the default argument for `begin()`.

**Comment by sjunges** [comment]
I see. Then I am ok with this solution, but would personally have considered seperating the implementation of the functions without argument.

The comment about the "end" code still stands? As in, shouldnt one be able to also call end(0) now?

**Comment by tquatmann** [comment]
I also think that `begin()` and `begin(index_type row)` should be two different overloads. Then, `begin(i)` is only valid iff row `i` exists and its the same for `end`.

**Comment by volkm** [comment]
Thanks for your comments. I decoupled `begin()` and `begin(row)`. This is also now consistent with `end` which already had both these functions.

**Comment by sjunges** [comment]
LGTM

**Pull request merged**